Allow comparison of two report directories

csordasmarton opened the pull request.

gyorb commented on libcodechecker/cmd/cmd_line_client.py:
```diff
@@ -391,6 +391,35 @@ def get_diff_base_results(client, baseids, base_hashes, suppressed_hashes):
+            for res in baseline_dir_results:
+                if res.report_hash not in new_hashes:
+                    filtered_reports.append(res)
+            print(base_hashes)
```
I think these prints were left from debugging.

csordasmarton commented:
@gyorb Done. I have also refactored the code a little bit. Now we are handling cases when basename parameter is a report directory and newname is a remote run. I have also added test cases to this.

gyorb commented on libcodechecker/cmd/cmd_line_client.py:
```diff
@@ -686,72 +853,77 @@ def get_run_tag(client, run_ids, tag_name):
+                                                args.basename,
+                                                os.path.abspath(args.newname))
+        print_diff_results(reports)
+        LOG.info("Compared remote run %s and local report directory %s",
```
remote run can be multiple runs if a user gave a wildcard pattern like `run*`, the printed text should reflect that.
The same issue applies to other places where run names are used for comparison.

It could help if we would print out the matching run names in these cases.
